cineast-core/src/main/java/org/vitrivr/cineast/core/features/Superpixel.java

```diff
@@ -0,0 +1,256 @@
+    regionMemberCount.resize(numSegments);
+
+    ImageSegmentationOps.countRegionPixels(pixelToRegion, numSegments, regionMemberCount.data);
+    colorize.process(color, pixelToRegion, regionMemberCount, segmentColor);
+
+    // ATTENTION:
+    // When uncommenting the following line, the superpixels get colored according to the colors specified in ColorSuperpixel.java
+    // segmentColor = usePredefinedColorSet(segmentColor);
+
+    return VisualizeRegions.regionsColor(pixelToRegion, segmentColor, null);
+  }
+
+  /**
+   * Maps segment colors to predefined colors
+   *
+   * @param segmentColor mean segmentColor of different fields
+   * @return segmentColor according to the closest color in {@link org.vitrivr.cineast.core.features.ColorSuperpixel}
+   */
+  private static FastQueue<float[]> usePredefinedColorSet(FastQueue<float[]> segmentColor) {
+    ColorSuperpixel colorSuperpixel = new ColorSuperpixel();
+    for (int i = 0; i < segmentColor.size; i++) {
+      double minDeltaE = Double.MAX_VALUE;
+      RGBContainer minColor = null;
+      LabContainer c1 = ColorConverter.RGBtoLab((int) segmentColor.data[i][0],
+          (int) segmentColor.data[i][1], (int) segmentColor.data[i][2]);
+      for (int j = 0; j < colorSuperpixel.colors.size(); j++) {
+        LabContainer c2 = ColorConverter.RGBtoLab(colorSuperpixel.colors.get(j).getR(),
+            colorSuperpixel.colors.get(j).getG(), colorSuperpixel.colors.get(j).getB());
+        double deltaE = calculateDeltaE(c1, c2);
+        if (deltaE < minDeltaE) {
+          minDeltaE = deltaE;
+          minColor = colorSuperpixel.colors.get(j);
+        }
+
+      }
+      assert minColor != null;
+      segmentColor.data[i][0] = minColor.getR();
+      segmentColor.data[i][1] = minColor.getG();
+      segmentColor.data[i][2] = minColor.getB();
+    }
+    return segmentColor;
+  }
+
+
+  /**
+   * Calculates delta E of two colours
+   *
+   * @param l1 colour 1
+   * @param l2 colour 2
+   * @return colour difference of l1 and l2
+   */
+  public static double calculateDeltaE(LabContainer l1, LabContainer l2) {
+
+    float dL = l1.getL() - l2.getL();
+    float dA = l1.getA() - l2.getA();
+    float dB = l1.getB() - l2.getB();
+
+    // calculate deltaE
+    return Math.sqrt(Math.pow(dL, 2) + Math.pow(dA, 2) + Math.pow(dB, 2));
+  }
+
+  /**
+   * This method applies the choice of the frame and calls computation of superpixel algorithms to a
+   * BufferedImage
+   *
+   * @param image original image to apply superpixel algorithms
+   * @param algID integer indicating which algorithm should be applied
+   * @return BufferedImage containing superpixels
+   */
+  public static BufferedImage applySuperpixelBI(BufferedImage image, int algID) {
+    BufferedImage mask = image;
+    image = ConvertBufferedImage.stripAlphaChannel(image);
+    ImageType<Planar<GrayF32>> imageType = ImageType.pl(3, GrayF32.class);
+
+    ImageSuperpixels alg;
+    switch (algID) {
```

Comment:
Why would you even switch on an id, rather than having different implementations of the same interface?